### PR TITLE
update plugin to work with latest payload

### PR DESCRIPTION
**Pull request by irhtml**
Note: I'm not to familiar with this plugin so not 100% sure I broke something, I tested with the supplied dev env and google auth and I manage to login.

Fixes https://github.com/WilsonLe/payload-oauth2/issues/11

Also fixes a few more breaking changes in the latest payload version

### CODE REVIEW

**Comment by tyteen4a03** [comment]
I tried installing your branch and couldn't get it to work - still redirecting me to failed login page.

**Comment by irhtml** [comment]
@tyteen4a03 do you get the same error as in #11  ?

TypeError: Cannot read properties of undefined (reading 'cookies')
    at generatePayloadCookie (webpack-internal:///(rsc)/../node_modules/.pnpm/payload@3.0.0-beta.112_graphql@16.9.0_monaco-editor@0.52.0_react-dom@19.0.0-rc-3edc000d-20240_incofvvsjgqtjnxeucoufluvym/node_modules/payload/dist/auth/cookies.js:74:50)
    at Object.handler (webpack-internal:///(rsc)/./src/oauth2/callback-endpoint.ts:152:94)
    ```

**Comment by MatthiasGrandl** [comment]
I tested this PR with beta.116 and it works for me.

**Comment by irhtml** [comment]
Updated to latest beta-116 @WilsonLe any input here? `1.0.3` is broken with latest beta.

File: src/callback-endpoint.ts [hunk 1]
```diff
@@ -165,8 +165,8 @@ export const createCallbackEndpoint = (
       // generate and set cookie
       // /////////////////////////////////////
       const cookie = generatePayloadCookie({
-        collectionConfig,
-        payload: req.payload,
+        collectionAuthConfig: collectionConfig.auth,
+        cookiePrefix: "payload",
```
Comment (WilsonLe) [inline src/callback-endpoint.ts:169]: This can be pulled from existing payload config rather than hard coding.

**Review by WilsonLe — CHANGES_REQUESTED** [review]

File: src/callback-endpoint.ts (hunk 1 above)
Comment (irhtml) [reply src/callback-endpoint.ts]: Ah, did not think of that. Fixed.

**Pull request merged**